include/boost/geometry/algorithms/detail/overlay/get_clusters.hpp

```diff
@@ -0,0 +1,302 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2021 Barend Gehrels, Amsterdam, the Netherlands.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_ALGORITHMS_DETAIL_OVERLAY_GET_CLUSTERS_HPP
+#define BOOST_GEOMETRY_ALGORITHMS_DETAIL_OVERLAY_GET_CLUSTERS_HPP
+
+#include <algorithm>
+#include <vector>
+
+#include <boost/geometry/algorithms/detail/overlay/approximately_equals.hpp>
+#include <boost/geometry/algorithms/detail/overlay/cluster_info.hpp>
+#include <boost/geometry/algorithms/detail/overlay/get_ring.hpp>
+#include <boost/range/value_type.hpp>
+#include <boost/geometry/util/math.hpp>
+
+#define BOOST_GEOMETRY_USE_RESCALING_IN_GET_CLUSTERS
+
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail { namespace overlay
+{
+
+template <typename T>
+struct sweep_equal_policy
+{
+    template <typename P>
+    static inline bool equals(P const& p1, P const& p2)
+    {
+        // Using machine epsilon results in too much points considered as inequal
+        return approximately_equals(p1, p2, 1.0); // 1000
+    }
+
+    static inline bool exceeds(T value)
+    {
+        // Its value is should be LARGER than the used kilo-epsilon
+        return value > 1.0e-3;
+    }
+
+};
+
+#ifdef BOOST_GEOMETRY_USE_RESCALING_IN_GET_CLUSTERS
+template <>
+struct sweep_equal_policy<boost::long_long_type>
```

Comment:
So this is a specialization for the type used for rescaling. What if a user passed a geometry using integral coordinates with rescaling disabled? There are bigger integral types than long long including e.g. MultiPrecision. My point is, shouldn't this be done for all integral coordinates?

Reply:
No, it's made for the rescaled integral grid.
But indeed, other integral coordinates should return "exactly equal".